unstructured/file_utils/model.py

```diff
@@ -41,8 +57,150 @@ def __lt__(self, other: FileType) -> bool:
         """
         return self.name < other.name
 
-    BMP = ("bmp", [".bmp"], "image/bmp", cast(list[str], []))
+    @classmethod
+    def from_extension(cls, extension: str | None) -> FileType | None:
+        """Select a FileType member based on an extension.
+
+        `extension` must include the leading period, like `".pdf"`. Extension is suitable as a
+        secondary file-type identification method but is unreliable for primary identification..
+
+        Returns `None` when `extension` is not registered for any supported file-type.
+        """
+        if extension in (None, "", "."):
+            return None
+        # -- not super efficient but plenty fast enough for once-or-twice-per-file use and avoids
+        # -- limitations on defining a class variable on an Enum.
+        for m in cls.__members__.values():
+            if extension in m._extensions:
+                return m
+        return None
+
+    @classmethod
+    def from_mime_type(cls, mime_type: str) -> FileType | None:
+        """Select a FileType member based on a MIME-type.
+
+        `extension` must include the leading period, like `".pdf"`. Extension is suitable as a
+        secondary file-type identification method but is unreliable for primary identification..
+        """
+        # -- not super efficient but plenty fast enough for once-or-twice-per-file use and avoids
+        # -- limitations on defining a class variable on an Enum.
+        for m in cls.__members__.values():
+            if mime_type == m._canonical_mime_type or mime_type in m._alias_mime_types:
+                return m
+        return None
+
+    @property
```

Comment:
Why use `@property` instead of `@lazyproperty` here?

Reply:
Mostly I guess that the computation on these are very modest and I didn't want to reason through the implications of `@lazyproperty` on an `enum.Enum` object, which has a lot of magic of its own going on. I'm sure it would be fine to make these lazyproperties but just didn't want to test it out :)